proto/redpanda/api/dataplane/v1/topic.proto

```diff
@@ -400,6 +498,68 @@ service TopicService {
       api: API_KAFKA
     };
   }
+  rpc ListTopicsConfigurations(ListTopicsConfigurationsRequest) returns (ListTopicsConfigurationsResponse) {
+    option (google.api.http) = {get: "/v1/topics/configurations"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List topics with configurations"
+      description: "List topics with their configurations, partition count and replication factor. Optional: filter based on topic name."
```

Comment:
partition count and replication factor aren't included here. The "filter" is not really a filter but more a selector and thus not optional I guess?